src/Epr.Reprocessor.Exporter.UI/Controllers/ReprocessingInputsAndOutputsController.cs

```diff
@@ -547,15 +632,26 @@ private async Task<IEnumerable<OrganisationPerson>> GetOrganisationPersons(UserD
 
     private static void SetJourneyForApplicationContactName(ReprocessorRegistrationSession session)
     {
-        var materialCount = session.RegistrationApplicationSession.ReprocessingInputsAndOutputs.Materials.Count;
-
-        if (materialCount == 1)
+        var materials = GetMaterials(session);
+        var lastUncheckedMaterial = materials?.Where(x => x.IsMaterialBeingAppliedFor == false).OrderBy(x => x.MaterialLookup.DisplayText).LastOrDefault();
+        if(lastUncheckedMaterial == null)
         {
-            session.Journey = [PagePaths.TaskList, PagePaths.ApplicationContactName];
+            if (materials?.Count == 1)
+            {
+                session.Journey = [PagePaths.TaskList, PagePaths.ApplicationContactName];
+            }
+            else
+            {
+                session.Journey = [PagePaths.PackagingWasteWillReprocess, PagePaths.ApplicationContactName];
+            }
         }
         else
         {
-            session.Journey = [PagePaths.PackagingWasteWillReprocess, PagePaths.ApplicationContactName];
+            var backPath = $"{PagePaths.MaterialNotReprocessingReason}?materialId={lastUncheckedMaterial.Id}";
+            session.Journey = [backPath, PagePaths.ApplicationContactName];
         }
     }
+
+    private static List<RegistrationMaterialDto> GetMaterials(ReprocessorRegistrationSession session) => session?.RegistrationApplicationSession?.ReprocessingInputsAndOutputs?.Materials;
+    private static List<RegistrationMaterialDto> GetAllUncheckedMaterials(ReprocessorRegistrationSession session) => GetMaterials(session)?.Where(x => x.IsMaterialBeingAppliedFor == false).OrderBy(x => x.MaterialLookup.DisplayText).ToList();
```

Comment:
Resharper is suggesting that these null checks are not required:
```suggestion
private static List<RegistrationMaterialDto> GetMaterials(ReprocessorRegistrationSession session) => session.RegistrationApplicationSession.ReprocessingInputsAndOutputs.Materials;
private static List<RegistrationMaterialDto> GetAllUncheckedMaterials(ReprocessorRegistrationSession session) => GetMaterials(session).Where(x => x.IsMaterialBeingAppliedFor == false).OrderBy(x => x.MaterialLookup.DisplayText).ToList();
```